frontend/src/functions/cookies.ts

```diff
@@ -41,3 +43,17 @@ export function clearCookie(cookieId: string) {
 export function getCsrfToken(): string | undefined {
   return getCookie("csrftoken");
 }
+
+export function convertUtmCookieToGaField(
+  utmCookie: string,
+): ReactGa.FieldsObject {
+  const campaignField = utmCookie.split("=");
+  const campaignFieldName = campaignField[0].replace("utm_", "");
+  const capitalizedCampaignFieldName =
+    campaignFieldName.charAt(0).toUpperCase() + campaignFieldName.slice(1);
+  const campaignFieldValue = campaignField[1];
+  const gaField: ReactGa.FieldsObject = {};
+  const gaFieldName = `campaign${capitalizedCampaignFieldName}`;
+  gaField[gaFieldName] = campaignFieldValue;
+  return gaField;
+}
```

Comment:
suggestion (non-blocking): This seems pretty efficient to me and readable. As Vincent mentioned, perhaps adding a comment for the input/output expected would be useful. We could add a test later.

Reply:
Added comment.